core/storage/question/gae_models.py

```diff
@@ -316,25 +316,70 @@ def get_question_skill_links_by_skill_ids(
         ) * question_count
 
         if not start_cursor == '':
-            cursor = datastore_query.Cursor(urlsafe=start_cursor)
-            question_skill_link_models, next_cursor, more = cls.query(
-                cls.skill_id.IN(skill_ids)
-                # Order by cls.key is needed alongside cls.last_updated so as to
-                # resolve conflicts, if any.
-                # Reference SO link: https://stackoverflow.com/q/12449197
-            ).order(-cls.last_updated, cls.key).fetch_page(
-                question_skill_count,
-                start_cursor=cursor
-            )
+            direction = start_cursor[-1]
```

Comment:
This is really messy and has a lot of duplicated code. Please try and consolidate it more, and then I'll take another pass. Try to follow the DRY principle ("don't repeat yourself") -- this is very important for writing maintainable code.

Reply:
I am not sure if it is possible to consolidate because for each case the query conditions are different.

Reply:
There are two invocations of ">" and two invocations of "<=". Each of these are repeats and it should still be possible to consolidate. You might need to create additional functions that you can reuse.

Reply:
I was able to reduce one such instance but I feel that the code not easy to understand. The other instance of  > has a difference in cursor which has to be in if condition so there has to be two instances.

Reply:
You can create a helper method to do this:
Something on the lines of: 
```
def get_question_order(direction, question_id, random_id):
    return question_id > random_id if direction == 1 else question_id <= random_id

def get_questions_list(skill_ids, random_id, question_skill_count, direction, cursor):
return cls.query(  cls.skill_id.IN(skill_ids)
                ).order(cls.question_id, cls.key).filter(
                    get_question_order(direction, cls.question_id, random_id)
                ).fetch_page(
                    question_skill_count,
                    start_cursor=cursor
                )
```

And then you can just call the helper function as explained below